### PR TITLE
address2 issues fixed

### PR DESCRIPTION
**Pull request by bbqcandy**
Fixed the issues with address2 in `database.rs` and `routes.rs`
tests all should pass as well

### CODE REVIEW

File: src/backend/src/database.rs [hunk 1]
```diff
@@ -499,41 +499,41 @@ mod tests {
...
+        let success_store = crate::database::update_profile(&pool, &user_id, full_name, address1, &address2, city, state, zip).await;
         
         assert!(success_store.is_ok());
 
         // test get_profile
         let profile = crate::database::get_profile(&pool, &user_id).await.unwrap().unwrap();
         assert_eq!(profile.full_name, full_name);
         assert_eq!(profile.address1, address1);
-        assert_eq!(profile.address2.unwrap(), address2);
+        assert_eq!(profile.address2.as_ref().unwrap_or(&String::new()), address2.as_ref().unwrap_or(&String::new()));
```
Comment (Brod8362) [inline src/backend/src/database.rs:515]: I think there's a better way to do this, this seems very excessive.
Should be as easy as `assert_eq!(address2, profile.address2)`

File: src/backend/src/database.rs [hunk 2]
```diff
@@ -499,41 +499,41 @@ mod tests {
...
+        let success_store = crate::database::update_profile(&pool, &user_id, full_name, address1, &address2, city, state, zip).await;
 
         assert!(success_store.is_ok());
 
         // test get_profile
         let profile = crate::database::get_profile(&pool, &user_id).await.unwrap().unwrap();
         assert_eq!(profile.full_name, full_name);
         assert_eq!(profile.address1, address1);
-        assert_eq!(profile.address2.unwrap(), address2);
+        assert_eq!(profile.address2.as_ref().unwrap_or(&String::new()), address2.as_ref().unwrap_or(&String::new()));
```
Comment (Brod8362) [inline src/backend/src/database.rs:536]: There should be a better way (see other comment)

File: src/backend/src/routes.rs [hunk 3]
```diff
@@ -201,15 +201,14 @@ pub async fn profile_submit(ctx: &UserContext, pool: &State<Pool<Sqlite>>, form:
             )
         }
     }
-    
-    // if address2 is empty, set it to an empty string
+    // if address2 is empty, set it to None
     let address2 = match &form.address2 {
```
Comment (Brod8362) [inline src/backend/src/routes.rs:205]: This shouldn't be needed anymore at all, can just pass it through directly from the form struct

**Review by Brod8362 — CHANGES_REQUESTED** [review]
See comments

**Comment by bbqcandy** [comment]
just made the changes requested

**Review by Mnguyener — APPROVED** [review]
Looks good

**Review by Brod8362 — CHANGES_REQUESTED** [review]
See single comment, otherwise good

**Comment by Brod8362** [comment]
Also, please have more descriptive commit messages that aren't just "fixed this" or "made changes"

**Pull request merged**